lib/tests.c

```diff
@@ -3211,11 +3211,18 @@ main(void)
         {"Test error messages", test_strerror},
         CU_TEST_INFO_NULL,
     };
+
+    /* We use initialisers here as the struct definitions change between
```

Comment:
Excellent! I'm going to merge this into master now, it's all fairly stable and seems to work well enough. You should be able to rebase your work off master after that, and it should all go smoothly from there. Give me a shout if not.